sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/AmqpClientOptions.java

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.amqp;
+
+import com.azure.core.util.ClientOptions;
+import com.azure.core.util.CoreUtils;
+import com.azure.core.util.Header;
+
+import java.util.Objects;
+import java.util.UUID;
+
+/**
+ * A set of AMQP client options.
+ */
+public class AmqpClientOptions extends ClientOptions {
+    private String identifier;
+
+    /** {@inheritDoc} **/
+    @Override
+    public ClientOptions setApplicationId(String applicationId) {
+        super.setApplicationId(applicationId);
+        return this;
+    }
+
+    /** {@inheritDoc} **/
+    @Override
+    public ClientOptions setHeaders(Iterable<Header> headers) {
+        super.setHeaders(headers);
+        return this;
+    }
+
+    /**
+     * Gets the identifier for the amqp client.
+     * @return Amqp client identifier.
+     */
+    public String getIdentifier() {
+        return CoreUtils.isNullOrEmpty(identifier) ? UUID.randomUUID().toString() : identifier;
```

Comment:
We should not generate a new identifier each time getIdentifier is called. Ideally, the business logic to generate an identifier should be on the client itself and not in the options type as this options type can be passed around to multiple client builders.

Reply:
Got it.